app/js/features/license/saga.js

```diff
@@ -32,29 +35,61 @@ export function *doCheckLicenseOwner() {
     const isLicenseOwner = yield call(License.methods.isLicenseOwner(web3.eth.defaultAccount).call);
     yield put({type: CHECK_LICENSE_OWNER_SUCCEEDED, isLicenseOwner});
   } catch (error) {
-    yield put({type: CHECK_LICENSE_OWNER_FAILED, error});
+    console.error(error);
+    yield put({type: CHECK_LICENSE_OWNER_FAILED, error: error.message});
   }
 }
 
 export function *onCheckLicenseOwner() {
   yield takeEvery(CHECK_LICENSE_OWNER, doCheckLicenseOwner);
 }
 
-export async function *doUserRating() {
+export function *doGetLicenseOwners() {
+  try {
+    // TODO get more information like position and rate
+    const events = yield License.getPastEvents('Bought', {fromBlock: 1});
```

Comment:
Just for my own understanding, is it cool to get all events from block 1?
Isn't it slow if you have millions of blocks?

Would it not be better to have this info store in the contract?
Also you may have sell your license no?

Reply:
Right, makes sense. I know it was the approach suggested by Iuri, but I agree that it will probably be slow. Let's ask the other pros.